client/QuickStart/API_Util.cs

```diff
@@ -0,0 +1,410 @@
+                    // For more information, see https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview
+                    return CreateManagedIdentityConnectionString(clusterUrl);
+
+                case "AppKey":
+                    // Learn More: For information about how to procure an AAD Application,
+                    // see: https://docs.microsoft.com/azure/data-explorer/provision-azure-ad-app
+                    // TODO (config - optional): App ID & tenant, and App Key to authenticate with
+                    return new KustoConnectionStringBuilder(clusterUrl).WithAadApplicationKeyAuthentication(Environment.GetEnvironmentVariable("APP_ID"), Environment.GetEnvironmentVariable("APP_KEY"), Environment.GetEnvironmentVariable("APP_TENANT"));
+
+                case "AppCertificate":
+                    // Authenticate using a certificate file.
+                    return CreateAppCertificateConnectionString(clusterUrl, certificatePath, certificatePassword, applicationId, tenantId);
+
+                default:
+                    ErrorHandler($"Authentication mode '{authenticationMode}' is not supported");
+                    return null;
+            }
+        }
+
+        /// <summary>
+        /// Generates Kusto Connection String based on 'ManagedIdentity' Authentication Mode.
+        /// </summary>
+        /// <param name="clusterUrl">Url of cluster to connect to</param>
+        /// <returns>ManagedIdentity Kusto Connection String</returns>
+        public static KustoConnectionStringBuilder CreateManagedIdentityConnectionString(string clusterUrl)
+        {
+            // Connect using the system - or user-assigned managed identity (Azure service only)
+            // TODO (config - optional): Managed identity client ID if you are using a user-assigned managed identity
+            var clientId = Environment.GetEnvironmentVariable("MANAGED_IDENTITY_CLIENT_ID");
+            return clientId is null ? new KustoConnectionStringBuilder(clusterUrl).WithAadSystemManagedIdentity() : new KustoConnectionStringBuilder(clusterUrl).WithAadUserManagedIdentity(clientId);
+        }
+
+        /// <summary>
+        /// Generates Kusto Connection String based on 'AppCertificate' Authentication Mode.
+        /// </summary>
+        /// <param name="clusterUrl">Url of cluster to connect to</param>
+        /// <param name="certificatePath">Given certificate path</param>
+        /// <param name="certificatePassword">Given certificate password</param>
+        /// <param name="applicationId">Given application id</param>
+        /// <param name="tenantId">Given tenant id</param>
+        /// <returns>AppCertificate Kusto Connection String</returns>
+        public static KustoConnectionStringBuilder CreateAppCertificateConnectionString(string clusterUrl, string certificatePath, string certificatePassword, string applicationId, string tenantId)
+        {
+            var missing = new[]
```

Comment:
I thought we are only doing issuer - you don't need these

Reply:
You do have thought the option to use the X509 certificate if publicCertFilePath is null. do we wish to remove it and stay only with issuer?

Reply:
Yes